modules/apps/headless/headless-site/headless-site-impl/src/main/java/com/liferay/headless/site/internal/resource/v1_0/SiteResourceImpl.java

```diff
@@ -463,15 +464,34 @@ private int _getMembershipRestriction(Integer membershipRestriction) {
 		return membershipRestriction;
 	}
 
-	private long _getParentGroupId(String parentSiteKey)
-		throws PortalException {
-
-		if (Validator.isNull(parentSiteKey)) {
+	private long _getParentGroupId(Group group, String parentSiteERC) {
+		if (Validator.isNull(parentSiteERC)) {
 			return GroupConstants.DEFAULT_PARENT_GROUP_ID;
 		}
 
-		Group parentGroup = _groupLocalService.getGroup(
-			contextCompany.getCompanyId(), parentSiteKey);
+		Group parentGroup =
+			_groupLocalService.fetchGroupByExternalReferenceCode(
+				parentSiteERC, contextCompany.getCompanyId());
+
+		if (LazyReferencingThreadLocal.isEnabled()) {
+			if (parentGroup == null) {
```

Comment:
This should be outside the lazy reference check to avoid an NPE on line 496 below (`return parentGroup.getGroupId();`). Unless the idea is to generate an NPE to avoid adding a group with an invalid parent group?

Reply:
In `SiteResourceTest.java` we have this test: `_testPostSiteFailureParentSiteNotFound()`, which expects the `postSite` to fail when it tries to add a site with a invalid `parentExternalReferenceCode`, so I think this is expected.